tests/commands/test_cmd_deploy.py

```diff
@@ -1,22 +1,71 @@
+# stdlib imports
+import os
+
 # third-party imports
-from click.testing import CliRunner
+import mock
+import responses
+
 
-# local imports
-from shpkpr.cli.entrypoint import cli
+def _test_template_path():
+    """Returns an absolute path to our test template
+    """
+    return os.path.abspath(os.path.join('tests', 'test.json.tmpl'))
 
 
-def test_no_args():
-    runner = CliRunner()
-    result = runner.invoke(cli, ['deploy'])
+def test_no_args(runner):
+    result = runner(['deploy'])
 
     assert result.exit_code == 2
     assert 'Usage:' in result.output
 
 
-def test_help():
-    runner = CliRunner()
-    result = runner.invoke(cli, ['deploy', '--help'])
+def test_help(runner):
+    result = runner(['deploy', '--help'])
 
     assert result.exit_code == 0
     assert 'Usage:' in result.output
     assert 'Deploy application from template.' in result.output
+
+
+@responses.activate
+@mock.patch('shpkpr.marathon.MarathonDeployment.wait')
+def test_no_force(mock_deployment_wait, runner, json_fixture):
+    responses.add(responses.PUT,
+                  'http://marathon.somedomain.com:8080/v2/apps/test-app',
+                  status=201,
+                  json=json_fixture("deployment"),
+                  match_querystring=True)
+    mock_deployment_wait.return_value = True
+
+    env = {
+        'SHPKPR_MARATHON_URL': "http://marathon.somedomain.com:8080",
+        'SHPKPR_APPLICATION': 'test-app',
+        'SHPKPR_DOCKER_REPOTAG': 'goexample/outyet:latest',
+        'SHPKPR_DOCKER_EXPOSED_PORT': '8080',
+        'SHPKPR_DEPLOY_DOMAIN': 'mydomain.com',
+    }
+    result = runner(['deploy', '--template', _test_template_path()], env=env)
+
+    assert result.exit_code == 0
+
+
+@responses.activate
+@mock.patch('shpkpr.marathon.MarathonDeployment.wait')
+def test_force(mock_deployment_wait, runner, json_fixture):
```

Comment:
Should this test (and maybe the other `test_force` cases) test that a forced deployment works after the Marathon API returns a 409?